apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -553,38 +565,34 @@ def _set_reader_coord_by_py_seq_or_np_array(
         # See libtiledbsoma.cc for more context on why we need the
         # explicit type-check here.
 
-        if dim.dtype == np.int64:
+        if pa.types.is_int64(dim.type):
             sr.set_dim_points_int64(dim.name, coord)
-        elif dim.dtype == np.int32:
+        elif pa.types.is_int32(dim.type):
             sr.set_dim_points_int32(dim.name, coord)
-        elif dim.dtype == np.int16:
+        elif pa.types.is_int16(dim.type):
             sr.set_dim_points_int16(dim.name, coord)
-        elif dim.dtype == np.int8:
+        elif pa.types.is_int8(dim.type):
             sr.set_dim_points_int8(dim.name, coord)
-
-        elif dim.dtype == np.uint64:
+        elif pa.types.is_uint64(dim.type):
             sr.set_dim_points_uint64(dim.name, coord)
-        elif dim.dtype == np.uint32:
+        elif pa.types.is_uint32(dim.type):
             sr.set_dim_points_uint32(dim.name, coord)
-        elif dim.dtype == np.uint16:
+        elif pa.types.is_uint16(dim.type):
             sr.set_dim_points_uint16(dim.name, coord)
-        elif dim.dtype == np.uint8:
+        elif pa.types.is_uint8(dim.type):
             sr.set_dim_points_uint8(dim.name, coord)
-
-        elif dim.dtype == np.float64:
+        elif pa.types.is_float64(dim.type):
             sr.set_dim_points_float64(dim.name, coord)
-        elif dim.dtype == np.float32:
+        elif pa.types.is_float32(dim.type):
             sr.set_dim_points_float32(dim.name, coord)
-
-        elif dim.dtype == "str" or dim.dtype == "bytes":
-            sr.set_dim_points_string_or_bytes(dim.name, coord)
-
         elif (
-            dim.dtype == "datetime64[s]"
-            or dim.dtype == "datetime64[ms]"
-            or dim.dtype == "datetime64[us]"
-            or dim.dtype == "datetime64[ns]"
+            pa.types.is_large_string(dim.type)
+            or pa.types.is_large_binary(dim.type)
+            or pa.types.is_string(dim.type)
+            or pa.types.is_binary(dim.type)
```

Comment:
maybe define an `is_pyarrow_anystr` (or a better name???) function for this? It appears a lot.